### ISSUE TITLE
Linux binaries using too-recent GLIBC

### ISSUE THREAD

**Issue by hannes** [r-universe-org/help#476]
I'm trying to run r-universe Linux binaries on Google Colab. This fails because the binaries link a relatively new GLIBC (as far as I can tell, version 2.38, which is only a little over a year old. 

It might be nice to intentionally use an older toolchain to build those binaries in the interest of platform compatibility, not just on Colab. For example, we at WC-Eend, I mean we at DuckDB link GLIBC 2.2.5 which is seven years old.

I should note that linking to an older GLIBC does not mean the binary will be slower, it will still *use* the newer GLIBC code where available, since this is only a symbolic link. But GLIBC uses an explicit symbol linking trick to ensure you're not running on an *older* version than linked.

**Comment by jeroen** [comment]
We indeed build and check binaries on `ubuntu:latest` docker container. This choice is based on the fact that many other R packages require somewhat recent system libraries (other than libc++). See https://r-universe.dev/sysdeps

As far as I can see google colab [currently uses ubuntu 22.04](https://medium.com/google-colab/colab-updated-to-ubuntu-22-04-lts-709a91555b3c).

We could add another build to the matrix that targets an older toolchain (e.g. based ubuntu 22.04, or rockylinux 8 or 9). That way you get somewhat portable binaries for packages like duckdb. 

But I want to point out that it does not solve the problem in general, because R packages that dynamically link to system libs will only run on the platform they were built on, because the ABI versions of external libraries change between distributions.

**Comment by hannes** [comment]
Hey @jeroen, thanks for getting back to me on this!

I would perhaps recommend looking at Python's `manylinux` images, they are particularly designed to create binaries that "just run" on many Linux distros while having a somewhat recent compiler. I've hacked something up that works for DuckDB's R package: https://gist.github.com/hannes/145e57bd3716430c6dad0f4a904f1de4

Here is a colab notebook that uses the resulting binary: https://colab.research.google.com/drive/1PxvkZ4FpMNtP-CpKpz5hvH-xKgaYC3-S#scrollTo=yREKPWZvoOx4

**Comment by jeroen** [comment]
> I would perhaps recommend looking at Python's manylinux images, they are particularly designed to create binaries that "just run" on many Linux distros while having a somewhat recent compiler. 

Right, manylinux is based on an older RHEL with backported compilers from the "RedHat developer toolset". These binaries "just run" as long as the R package does not link to external system libraries.

For example, if we want to build an R package which links to openssl, we first need `yum install openssl-devel`. Because this gets us openssl 1.1.1, the resulting binary won't run on ubuntu 22 or ubuntu 24, which have the openssl 3 runtime.
